### ISSUE TITLE
Cannot find an overload for "StartProcessAsCurrentUser" and the argument count: "7"

### ISSUE THREAD

**Issue by PJMarcum** [KelvinTegelaar/RunAsUser#31]
Can anyone please tell me what I am doing wrong to cause this:

PS C:\Windows\system32> Import-Module -Name runasuser

PS C:\Windows\system32> Invoke-AsCurrentUser -scriptblock $scriptblock -UseWindowsPowerShell -CaptureOutput 
Invoke-AsCurrentUser : Could not execute as currently logged on user: Cannot find an overload for "StartProcessAsCurrentUser" and the argument count: "7".
At line:1 char:1
+ Invoke-AsCurrentUser -scriptblock $scriptblock -UseWindowsPowerShell  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], MethodException
    + FullyQualifiedErrorId : System.Management.Automation.MethodException,Invoke-AsCurrentUser

PS C:\Windows\system32> whoami
nt authority\system

**Comment by DaleMontgomeryPFIT** [comment]
Have a similar problem on Windows 11 OS computers:
`invoke-ascurrentuser : Could not execute as currently logged on user: Exception calling "StartProcessAsCurrentUser" 
with "7" argument(s): "CreateProcessAsUser failed. (Access is denied, Win32ErrorCode 5 - 0x00000005)"
At C:\windows\temp\8x8-InstallAsUser.ps1:76 char:5
+     invoke-ascurrentuser -CaptureOutput -NonElevatedSession -ScriptBl ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], MethodInvocationException
    + FullyQualifiedErrorId : System.Management.Automation.MethodInvocationException,Invoke-AsCurrentUser
invoke-ascurrentuser : Could not execute as currently logged on user: Exception calling "StartProcessAsCurrentUser"
with "7" argument(s): "CreateProcessAsUser failed. (Access is denied, Win32ErrorCode 5 - 0x00000005)"`

**Comment by KelvinTegelaar** [comment]
Just a check;  is a user logged on locally or not? :)

**Comment by DaleMontgomeryPFIT** [comment]
in my case, yes

**Comment by PJMarcum** [comment]
I used an older version of the module and it works.

**Comment by KelvinTegelaar** [comment]
The older versions don't have -captureoutput and that's something you've put into the command, so you're most likely running a different command than. Please post the entire script you're trying to run.

**Comment by PJMarcum** [comment]
Right, I had to change the command. I'm using this now: invoke-ascurrentuser -NonElevatedSession -scriptblock $scriptblock -UseWindowsPowerShell

I can post the entire script but then you will hate me because my scripting is very bad. LOL

**Comment by DaleMontgomeryPFIT** [comment]
In my case it is version 2.3.1 and relating code is simple:

$scriptblock = {
        $InstallerPath = "c:\install_media\work-64-exe-v7.22.3-4.exe"
        Start-Process "$InstallerPath" -Wait
    }
    invoke-ascurrentuser -CaptureOutput -NonElevatedSession -ScriptBlock $scriptblock
    Write-Msg "completed invoke as user"
